detox/index.d.ts

```diff
@@ -1091,6 +1091,11 @@ declare global {
              */
             typeText(text: string): Promise<void>;
 
+            /**
+             * reset the state of the app (e.g. by reinstalling on the device)
+             */
+            resetAppState(): Promise<void>;
```

Comment:
On 2nd thought, I don't think that exposing this as an external API is the right way at all.
From the users' perspective, all they normally do (in order to relaunch the app from scratch) is to invoke `device.launchApp({delete: true})`. Whether the app gets reinstalled or has its state cleaned up should be agnostically decided [here](https://github.com/wix/Detox/blob/35b5bbc912e05dec9a8f5f1a6f8c03ec0290d883/detox/src/devices/runtime/RuntimeDevice.js#L318). This would make this whole optimization applied automatically and seamlessly into existing tests out there.

@noomorph I would highly appreciate you voicing your opinion on this as well 🙏🏻